hal/src/boron/network/sara_ncp_client.cpp

```diff
@@ -360,8 +361,12 @@ int SaraNcpClient::dataChannelWrite(int id, const uint8_t* data, size_t size) {
     if (err) {
         // Make sure we are going into an error state if muxer for some reason fails
         // to write into the data channel.
-        disable();
         connectionState(NcpConnectionState::DISCONNECTED);
+        // disable() sets the ncpState_ to NcpState::DISABLED which prevents connectionState()
+        // from even trying to send events to the PPP client to go to a LOWER_DOWN state,
+        // and it might still not get there if connState_ is already DISCONNECTED, but order matters.
+        // Call disable() after connectionState()
+        disable();
```

Comment:
It should be noted that B5 SoM does not contain this same error handling block.  Should we add it?

Reply:
We should. B5 SoM code was branched off of this one before the change was made.